packages/big-design/src/components/Dropdown/Dropdown.tsx

```diff
@@ -61,16 +61,42 @@ export const Dropdown = memo(
       [],
     );
 
+    const stateReducer: UseSelectProps<DropdownItem | DropdownLinkItem>['stateReducer'] = (
+      state,
+      actionAndChanges,
+    ) => {
+      const { changes, type } = actionAndChanges;
+
+      switch (type) {
+        case useSelect.stateChangeTypes.ToggleButtonKeyDownArrowDown:
+          if (state.highlightedIndex === flattenedItems.length - 1) {
+            return { ...changes, highlightedIndex: 0 };
+          }
+
+          return changes;
+
+        case useSelect.stateChangeTypes.ToggleButtonKeyDownArrowUp:
+          if (state.highlightedIndex === 0) {
+            return { ...changes, highlightedIndex: flattenedItems.length - 1 };
+          }
+
+          return changes;
+
+        default:
+          return changes;
+      }
+    };
```

Comment:
This replaces [`circularNavigation`](https://github.com/downshift-js/downshift/blob/master/src/hooks/MIGRATION_V7.md#circularnavigation) as part of their migration guide.